controller/cmd/public-api/main.go

```diff
@@ -46,7 +46,9 @@ func Main(args []string) {
 	defer destinationConn.Close()
 
 	k8sAPI, err := k8s.InitializeAPI(
-		*kubeConfigPath, true,
+		context.Background(),
```

Comment:
As commented elsewhere, it might make sense to use the same context as in the shutdown() call below